service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -122,21 +129,35 @@ String value() {
      * @param config The PrometheusBasedResourceLimitChecks configuration object.
      * @param cacheProvider The cache provider to use for creating caches for metrics data retrieved
      *                      from the prometheus backend or {@code null} if they should not be cached.
+     * @param tracer The tracer instance.
      * @throws NullPointerException if any of the parameters except cacheProvider are {@code null}.
      */
     public PrometheusBasedResourceLimitChecks(final WebClient webClient,
-            final PrometheusBasedResourceLimitChecksConfig config, final CacheProvider cacheProvider) {
+            final PrometheusBasedResourceLimitChecksConfig config, final CacheProvider cacheProvider, final Tracer tracer) {
```

Comment:
this is a non backward compatible change to public API. IMHO we need to overload the constructor instead ...

Reply:
yes, I have added an overloaded constructor and pushed the changes.